### PR TITLE
Getting rid of executeActions

### PR DESCRIPTION
**Pull request by alyssawilk**
@mattklein123 feel free to shoot this down if you find the closures easier to read but tests are easier for me to follow this way.

this does mean more initialize() but I've tried to compensate by automating cleanup.

#1518

### CODE REVIEW

File: test/integration/http_integration.cc [hunk 1]
```diff
@@ -274,32 +272,24 @@ void HttpIntegrationTest::testRouterRequestAndResponseWithBody(
...
-                    Http::TestHeaderMapImpl request_headers{{":method", "GET"},
-                                                            {":path", "/test/long/url"},
-                                                            {":scheme", "http"},
-                                                            {":authority", "host"},
-                                                            {"x-lyft-user-id", "123"}};
-                    sendRequestAndWaitForResponse(request_headers, 0, default_response_headers_, 0);
-                  },
-                  // Clean up downstream.
-                  [&]() -> void { codec_client_->close(); }});
+  if (!initialized_) {
```
Comment (htuch) [inline test/integration/http_integration.cc]: How come this special case?

File: test/integration/cors_filter_integration_test.cc [hunk 2]
```diff
@@ -20,30 +20,21 @@ class CorsFilterIntegrationTest : public HttpIntegrationTest,
...
 
   void testNormalRequest(Http::TestHeaderMapImpl&& request_headers,
                          Http::TestHeaderMapImpl&& expected_response_headers) {
-    executeActions({
-        [&]() -> void { codec_client_ = makeHttpConnection(lookupPort("http")); },
-        [&]() -> void {
-          sendRequestAndWaitForResponse(request_headers, 0, expected_response_headers, 0);
-        },
-        [&]() -> void { response_->waitForEndStream(); },
-        [&]() -> void { cleanupUpstreamAndDownstream(); },
```
Comment (htuch) [inline test/integration/cors_filter_integration_test.cc]: It seems in some places that cleanup has been dropped, but in others it's still around. Can you talk about what the guiding principle here is?

Comment (alyssawilk) [reply test/integration/cors_filter_integration_test.cc]: I think I missed one call in proto_integration_test.cc which I've fixed.  Please point out others you see.

**Comment by mattklein123** [comment]
Nice LGTM. Can you merge master to get Circle tests running? Also there is some failure on old Travis. Not sure what that is about.

**Comment by alyssawilk** [comment]
Yeah, it looked like a shutdown complaint from tsan.  I couldn't repro locally  but I believe the problem was that the server was shutting down in TearDown() then the cleanup was happening in the destructor.  This means the server was shutting down without the client/upstream being stopped which seems a likely cause of races.  By sticking them all in the same function I restored prior ordering and hopefully fixed the race.

**Comment by alyssawilk** [comment]
Oh, closing and reopening doesn't work for retriggering circleCI?  Curses.

FWIW failure appears unrelated to PR.  I wonder if a change to commit and submit -f will work.

Executed 159 out of 159 tests: 159 tests pass.
There were tests whose specified size is too big. Use the --test_verbose_timeout_warnings command line option to see which ones these are.
Copying for image build...
cp: cannot stat '/build/envoy/envoy/source/exe/envoy': No such file or directory
Exited with code 1

**Comment by mattklein123** [comment]
@alyssawilk sorry this fix is needed: https://github.com/envoyproxy/envoy/pull/1741. Then you will have to merge master again.

**Comment by alyssawilk** [comment]
Ah, thanks for the update.  CircleCI is so much faster though - it's going
to be awesome when we've worked the kinks out!

**Pull request merged**